tracer/test/test-applications/integrations/Samples.AzureServiceBus/README.md

```diff
@@ -0,0 +1,186 @@
+## Instrumentation behaviors
+Before reading on, make sure you understand the current instrumentation behavior to make sense of the generated traces from `Azure.Messaging.ServiceBus`:
+
+### ServiceBusSender.SendMessageAsync and ServiceBusSender.SendMessagesAsync
+When this API is called, the following diagnostic activity occurs:
+1. A `Message` Activity is started, whose TraceId/SpanId is injected into the Azure Service Bus message's properties `Diagnostic-Id` and `traceparent` (and `tracestate` property if applicable)
+2. A `ServiceBusSender.Send` Activity is started that represents the sending operation.
+3. The `ServiceBusSender.Send` Activity is decorated with a Span Link for each `Message` Activity
+
+### ServiceBusProcessor and ServiceBusSessionProcessor
+When `ServiceBusProcessor.StartProcessingAsync()` is called, it will continuously poll the desired queue for messages and handle them in a callback, which allows distributed tracing to work with a parent-child relationship. **However**, the polling generates a `ServiceBusReceiver.Receive` Activity outside of the trace context that is likely to be orphaned into its own trace.
+
+## Test application behavior
+The test application consists of 5 separate test methods, which will be explained individually.
+
+### TestServiceBusProcessorAsync
+Source: [`RequestHelper.Processor.cs`](./RequestHelper.Processor.cs)
+API tested: ServiceBusProcessor
+
+The test method initializes a ServiceBusProcessor, opens a root span, sends one message
+
+### TestServiceBusSessionReceiverAsync
+> Source: [`RequestHelper.Session.cs`](./RequestHelper.Session.cs)
+
+### TestSenderSchedulingAsync
+> Source: [`RequestHelper.cs`](./RequestHelper.cs)
+
+### TestServiceBusReceiverIndividualMessageAsync
+> Source: [`RequestHelper.cs`](./RequestHelper.cs)
+
+### TestServiceBusReceiverBatchMessagesAsync
+> Source: [`RequestHelper.cs`](./RequestHelper.cs)
+
+
+## Resulting Traces
+### TestServiceBusProcessorAsync traces
+Trace #1
+
+```plaintext
+SendMessageToProcessorAsync
+-----------
+|          |
+Message    ServiceBusSender.Send
+|
+ServiceBusProcessor.ProcessMessage
+|
+ServiceBusReceiver.Complete
+```
+
+Trace #2
```

Comment:
TraceId_5 2

Reply:
Added this in 46542c7eccdecdbb612460fde446cbc4ce3c2cdf